efficientnetv2/main.py

```diff
@@ -464,35 +464,25 @@ def main(unused_argv):
           mixup_list = np.linspace(0, config.data.mixup_alpha, total_stages)
           cutmix_list = np.linspace(0, config.data.cutmix_alpha, total_stages)
 
-        ibase = config.data.ibase or (input_image_size / 2)
-        # isize_list = np.linspace(ibase, input_image_size, total_stages)
-        for stage in range(curr_step // train_steps, total_stages):
-          tf.compat.v1.reset_default_graph()
-          ratio = float(stage + 1) / float(total_stages)
-          max_steps = int(ratio * train_steps)
-          image_size = int(ibase + (input_image_size - ibase) * ratio)
-          params['image_size'] = image_size
-
-          if config.train.sched:
-            config.data.ram = ram_list[stage]
-            config.data.mixup_alpha = mixup_list[stage]
-            config.data.cutmix_alpha = cutmix_list[stage]
-            # config.model.dropout_rate = dp_list[stage]
-
-          ds_lab_cls = datasets.build_dataset_input(True, image_size,
-                                                    image_dtype, FLAGS.data_dir,
-                                                    train_split, config.data)
-
-          est = tf.estimator.tpu.TPUEstimator(
-              use_tpu=FLAGS.use_tpu,
-              model_fn=model_fn,
-              config=run_config,
-              train_batch_size=config.train.batch_size,
-              eval_batch_size=config.eval.batch_size,
-              export_to_tpu=FLAGS.export_to_tpu,
-              params=params)
-          est.train(
-              input_fn=ds_lab_cls.input_fn, max_steps=max_steps, hooks=hooks)
+        def _input_fn(params):
+          ibase = config.data.ibase or (input_image_size / 2)
+          isize_list = np.linspace(ibase, input_image_size, total_stages).astype(np.int32)
+          fns = {}
+          for index, image_size in enumerate(isize_list):
+            def _fn():
+              if config.train.sched:
+                config.data.ram = ram_list[index]
+                config.data.mixup_alpha = mixup_list[index]
+                config.data.cutmix_alpha = cutmix_list[index]
+              ds_lab_cls = datasets.build_dataset_input(True, image_size,
+                                                        image_dtype, FLAGS.data_dir,
+                                                        train_split, copy.deepcopy(config.data))
+              return ds_lab_cls.input_fn(params)
+
+            fns[index] = _fn
+          return tf.switch_case(tf.cast(tf.train.get_global_step() // train_steps, tf.int32), fns)
```

Comment:
Could you check whether the image size changes with the global step on TPU.

Reply:
Hmm, looks like it has some issues on TPU, maybe it is because TF1 doesn't do JIT compile and needs static shape at the first time of complication.

Could you help add it to main_tf2.py?  TF2 script can JIT compilation and can probably support this. I plan to make main_tf2.py as default.

Reply:
It's a bug in switch_case(https://github.com/tensorflow/tensorflow/issues/49378), which happens on tf2 too.